### PR TITLE
Implement Node Reboot Storage API

### PR DESCRIPTION
**Pull request by yokaze**
This pull request implements the data struct and the storage API for node reboot functionality.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ const (
 	KeyClusterRevision       = "cluster-revision"
 	KeyConstraints           = "constraints"
 	KeyLeader                = "leader/"
+	KeyReboots               = "reboots/data/"
```
Comment (ymmt2005) [inline storage.go:33]:
```suggestion
	KeyRebootsPrefix               = "reboots/data/"
```

File: storage.go [hunk 2]
```diff
@@ -685,6 +687,178 @@ func (s Storage) GetSabakanURL(ctx context.Context) (string, error) {
...
+		return err
+	}
+	if !txnResp.Succeeded {
+		goto RETRY
+	}
+
+	return nil
+}
+
+// GetRebootEntry loads the entry specified by the index from the reboot queue.
```
Comment (ymmt2005) [inline storage.go:775]:
```suggestion
// GetRebootsEntry loads the entry specified by the index from the reboot queue.
```

**Comment by morimoto-cybozu** [comment]
@ymmt2005 
Review again please.
I'll squash the commits if OK.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Review by morimoto-cybozu — APPROVED** [review]
Just tidy up by squashing after ymmt2005's LGTM.

**Pull request merged**